### PR TITLE
Add partition handling to db schema test

### PR DESCRIPTION
**Pull request by gregflack**
- add sql queries for tables partition information
- add parser for partition information
- update schema test

### CODE REVIEW

File: integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java [hunk 1]
```diff
@@ -52,32 +53,55 @@ private TestDataHelper() {}
    */
   public static void generateDbWithTables(Connection connection, int tableCount)
       throws SQLException, IOException {
-    String createDb = SqlHelper.getSql(Constants.SQL_TEST_DATA_BASE_PATH + "create_db.sql");
-    String createTable = SqlHelper.getSql(Constants.SQL_TEST_DATA_BASE_PATH + "create_table.sql");
+    String createDb = SqlHelper.getSql(SQL_TEST_DATA_BASE_PATH + "create_db.sql");
+    String createTable = SqlHelper.getSql(SQL_TEST_DATA_BASE_PATH + "create_table.sql");
+    String populateData = SqlHelper.getSql(SQL_TEST_DATA_BASE_PATH + "populate_data.sql");
```
Comment (haselbach) [inline integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java:58]: So is this a format string? If so, maybe rename it accordingly.

File: integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java [hunk 2]
```diff
@@ -17,11 +17,12 @@
 
 import static com.google.cloud.bigquery.dwhassessment.base.Constants.DB_SCHEMAS_FILE;
 import static com.google.cloud.bigquery.dwhassessment.base.Constants.SCHEMAS_PATH;
+import static com.google.cloud.bigquery.dwhassessment.base.Constants.SQL_TEST_DATA_BASE_PATH;
```
[the remark this answers is not in the document]
Comment (gregflack) [reply integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java:20]: You have fight but partialy. Constant classes can be splited to particular context so they will be initialize with this context. I can split this class to more concrete constant context classes. Another approach to store constants is interface, but this is pure antipatern - lack of control in inheritence (constants start to be inherited in inheritence tree below they was introduced). What do you thing - Should I split this constant class to more entities like data generation, database, test classes or do you have another approach.

File: integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java (hunk 1 above)
Comment (gregflack) [reply integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java:58]: No, it isn't format string getSql method reads content of sql files

File: integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/pojo/TestData.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.Random;
+
+public class TestData {
+
+  private final int id;
+  private final String name;
+  private final String description;
+  private final String state;
+
+  public TestData() {
```
[the remark this answers is not in the document]
Comment (gregflack) [reply integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/pojo/TestData.java:29]: Yes I know, but for one such simle entity I don't see reason to introduce addtional dependencies in POM file, what do you think?

[the remark this answers is not in the document]
Comment (haselbach) [reply integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/pojo/TestData.java:30]: Fair enough. I thought there was already a dependency on that.

File: integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (haselbach) [reply integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java:20]: We can leave it as is for now. IMHO constants should be put into the context of some class that can take reasonable ownership of it.

File: integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java (hunk 1 above)
Comment (haselbach) [reply integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/testdata/TestDataHelper.java:58]: Ah, yes, I got confused by the usage of format but I now remember that the SQL is really a format string that gets rendered to SQL. That's alright then.

**Review by haselbach — APPROVED** [review]

**Pull request merged**